logger/xp.py

```diff
@@ -144,14 +145,25 @@ def log_config(self, config_dict, to_visdom=True):
         if to_visdom and self.use_visdom:
             self.plotter.plot_config(config_dict)
 
-    def log_with_tag(self, tag, idx=None):
+    def log_with_tag(self, pattern, idx=None, reset=False):
+        """ Log metrics from each tag matching the given pattern.
+        Pattern parameter must be in Unix shell-style wildcards format.
+        """
+        # log metrics of all tags matching the given pattern
+        for tag in fnmatch.filter(self.metrics, pattern):
+            self._log_with_tag(tag, idx, reset)
+
+    def _log_with_tag(self, tag, idx=None, reset=False):
```

Comment:
It is confusing to have two separate functions with such similar names. Can you fuse them into a single function, where the `tag` argument is simply processed with `fnmatch` as a first step?

Reply:
Also, let's insert a check that the tag (processed after fnmatch) actually exists, I don't think a user would input some tags that are not expected to match with existing metrics